src/libraries/Microsoft.PowerFx.Core/Public/Values/OptionSetValue.cs

```diff
@@ -61,6 +61,12 @@ public string DisplayName
 
                 return logicalName;
             }
-        }        
+        }
+
+        public override string ToExpression()
+        {
+            // !JYL! Do we need to serialize this?
```

Comment:
>/ !JYL! Do we need to serialize this? [](http://example.com/codeflow?start=13&length=37)

Yes :)

Reply:
Yes, would need to be `$"{Type._type.OptionSetInfo.EntityName}.{Option}`, the host would be responsible for making sure the option set is registered under the same name

Reply:
What about $"{Type._type.OptionSetInfo.EntityName}.{**_DisplayName_**}"?

Reply:
We always serialize in the invariant locale (logical names), so we'd want to use Option in this case. The host may provide different display names when we're running this, but the logical names would be the same.